sympl/_core/restore_dataarray.py

```diff
@@ -8,9 +8,10 @@
 
 
 def ensure_values_are_arrays(array_dict):
-    for name, value in array_dict.items():
-        if not isinstance(value, np.ndarray):
-            array_dict[name] = np.asarray(value)
+    pass
```

Comment:
This is a temporary and dirty solution. We could think of a mechanism to control whether arrays must be coerced or not.

Reply:
The `asarray` function of Numpy seeks to coerce the input array-like storage `value` into a `ndarray`. This operation could break e.g. the data layout and the memory alignment of `value`. In the specific case of Tasmania, `value` could be a GT4Py storage, whose low-level details and features are fitted to the target computing architecture and thus must be preserved. The problem can be circumvented by monkey-patching Numpy via the function `gt4py.storage.prepare_numpy()`, but this is much GT4Py-specific. We could think of a more organic solution, or just pass `value` to `DataArray` as it is and let `DataArray` perform all type checks (and eventually throw exceptions).